azurerm/internal/services/containers/kubernetes_addons.go

```diff
@@ -84,6 +87,39 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
 					},
 				},
 
+				"ingress_appgw": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:     schema.TypeBool,
+								Required: true,
+							},
+							"appgw_id": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: azureHelpers.ValidateResourceID,
+							},
+							"appgw_name": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"appgw_subnet_id": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: azureHelpers.ValidateResourceID,
+							},
+							"appgw_subnet_prefix": {
```

Comment:
```suggestion
							"subnet_prefix": {
```
also is this something we could load from the gateway? or does it potentially have multiple?